### PR TITLE
Vijayasree/Refactor footer ts

### PR DESCRIPTION
**Pull request by vijayasree-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by alamin-deriv — APPROVED** [review]
LGTM

**Review by aswathy-deriv — APPROVED** [review]
LGTM

**Comment by NikitK-deriv** [comment]
And please try to fix code smells as well if it possible

File: src/components/custom/_eu-redirect-modal.tsx [hunk 1]
```diff
@@ -8,6 +7,17 @@ import device from 'themes/device.js'
...
 
+type EURedirectProps = {
+    aria_label?: string
+    closeModal?: () => void
+    is_open: boolean | (() => void)
+    ref?: Ref<HTMLAnchorElement>
+    rel?: string
+    target?: string
+    to?: string
+    toggle: boolean | (() => void)
```
Comment (prince-deriv) [inline src/components/custom/_eu-redirect-modal.tsx:18]: `boolean | (() => void)` this is being use quite often. Can you create a type for this under `types/generics` the type name can be `BooleanFuncVoidType`

File: src/components/custom/_eu-redirect-modal.tsx [hunk 2]
```diff
@@ -85,7 +95,16 @@ const Cancel = styled.span`
...
-const EURedirect = ({ toggle, is_open, closeModal, to, target, rel, ref, aria_label }) => {
+const EURedirect = ({
+    toggle,
+    is_open,
+    closeModal,
+    to,
+    target,
+    rel,
+    ref,
+    aria_label,
```
Comment (prince-deriv) [inline src/components/custom/_eu-redirect-modal.tsx:106]: Let's add default value for optional props

```suggestion
    toggle,
    is_open,
    closeModal,
    to = '',
    target = '',
    rel = '',
    ref,
    aria_label = '',
```

File: src/components/layout/layout.tsx [hunk 3]
```diff
@@ -266,30 +286,14 @@ const Layout = ({
...
-                to={modal_payload.to}
-                target={modal_payload.target}
-                rel={modal_payload.rel}
-                ref={modal_payload.ref}
-                aria_label={modal_payload.aria_label}
+                to={modal_payload['to']}
+                target={modal_payload['target']}
+                rel={modal_payload['rel']}
+                ref={modal_payload['ref']}
+                aria_label={modal_payload['aria_label']}
```
Comment (sean-binary) [inline src/components/layout/layout.tsx:293]: Were these flagged up by TS? @vijayasree-deriv

File: src/components/layout/location-context.tsx [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
-import PropTypes from 'prop-types'
+import React, { ReactNode, Dispatch, SetStateAction } from 'react'
 
-export const LocationContext = React.createContext(true) // undefined => true
+type LocationProviderProps = {
+    children: ReactNode
+    has_mounted?: boolean
+    setModalPayload?: Dispatch<SetStateAction<boolean>>
+    show_cookie_banner?: boolean
+    toggleModal?: boolean | (() => void)
```
Comment (sean-binary) [inline src/components/layout/location-context.tsx:8]: Same here.

File: src/components/layout/layout.tsx (hunk 3 above)
Comment (vijayasree-deriv) [reply src/components/layout/layout.tsx:293]: yes

**Review by mohammad-rework — APPROVED** [review]

**Review by aswathy-deriv — DISMISSED** [review]

**Review by sean-binary — APPROVED** [review]

**Review by prince-deriv — DISMISSED** [review]

**Review by mohammad-rework — APPROVED** [review]

**Review by NikitK-deriv — APPROVED** [review]

**Pull request closed without merging**